src/EditorFeatures/Core/RenameTracking/RenameTrackingTaggerProvider.RenameTrackingCommitter.cs

```diff
@@ -49,15 +49,14 @@ private class RenameTrackingCommitter
                 _renameSymbolResultGetter = new AsyncLazy<RenameTrackingSolutionSet>(c => RenameSymbolWorkerAsync(c), cacheResult: true);
             }
 
-            public void Commit(CancellationToken cancellationToken)
+            public async Task CommitAsync(CancellationToken cancellationToken)
             {
-                _stateMachine.ThreadingContext.ThrowIfNotOnUIThread();
-
-                ApplyChangesToWorkspace(cancellationToken);
+                await ApplyChangesToWorkspaceAsync(cancellationToken).ConfigureAwait(false);
 
                 // Clear the state machine so that future updates to the same token work,
                 // and any text changes caused by this update are not interpreted as 
                 // potential renames
+                await _stateMachine.ThreadingContext.JoinableTaskFactory.SwitchToMainThreadAsync(cancellationToken);
```

Comment:
is this switch to prevent concurrent access to the `_stateMachine`?  Should we just lock it instead

Reply:
it's more that clearing the tracking session does a bunch of UI affinitized things.  so i was switching back to that for its needs there.  i tried pushgin the async deeper, but it got a little big.

Reply:
ah I didn't realize the clear did a bunch of UI stuff.  Makes sense then